### ISSUE TITLE
Panic on `std.record.map` with an invalid argument

### ISSUE THREAD

**Issue by thufschmitt** [tweag/nickel#1414]
**Describe the bug**

`std.record.map` expects a function of two arguments as its second argument (taking the field name and field value). Passing a one-argument function instead causes a panic rather than a proper blame.

**To Reproduce**

```console
$ nickel export <<< 'std.record.map (fun x => x) { x = 1 }' 
thread 'main' panicked at 'path_span: we pretty-printed and parsed again the type of a label, so it must have all of its position defined, but `ty_path::span` returned `None`', src/error.rs:1369:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Expected behavior**

Nice error message explaining how I screwed up :)

**Environment**
 - OS name + version: NixOS 23.05
 - Version of the code: a4fad2b9354ac0a2aaa6c16b1ff5ebf2a616e7da (built from the Flake)

**Additional context**

With `RUST_BACKTRACE=1`:

```console
  RUST_BACKTRACE=1 nickel export <<< 'std.record.map (fun x => x) { x = 1 }'                                                                       
thread 'main' panicked at 'path_span: we pretty-printed and parsed again the type of a label, so it must have all of its position defined, but `ty_
path::span` returned `None`', src/error.rs:1369:10                                                                                                 
stack backtrace:                                                                                                                                   
   0: rust_begin_unwind                                                                                                                            
   1: core::panicking::panic_fmt                                                                                                                   
   2: core::option::expect_failed                                                                                                                  
   3: nickel_lang::error::blame_error::path_span                                                                                                   
   4: nickel_lang::error::blame_error::blame_diagnostics                                                                                           
   5: <nickel_lang::error::EvalError as nickel_lang::error::IntoDiagnostics<codespan::file::FileId>>::into_diagnostics                             
   6: <nickel_lang::error::Error as nickel_lang::error::IntoDiagnostics<codespan::file::FileId>>::into_diagnostics                                 
   7: nickel_lang::program::Program<EC>::report                                                                                                    
   8: nickel::main                                                                             
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.                                                            
```

**Comment by yannham** [comment]
I think we fixed this bug as of yesterday :smile: cf #1406 and #1407. I confirm that the latest master doesn't show this issue on my end:

```
cargo run -p nickel-lang-cli -- export <<< 'std.record.map (fun x => x) { x = 1 }'        
    Finished dev [unoptimized + debuginfo] target(s) in 0.11s
     Running `target/debug/nickel export`
error: contract broken by the caller of `map`
     ┌─ <stdlib/std.ncl>:1797:32
     │
1797 │       : forall a b. (String -> a -> b) -> { _ : a } -> { _ : b }
     │                                ------ expected return type of a function provided by the caller
     │
     ┌─ <unknown> (generated by evaluation):1:1
     │
   1 │ "x"
     │ --- evaluated to this value
     │
     = This error may happen in the following situation:
           1. A function `f` is bound by a contract: e.g. `(Number -> Number) -> Number`.
           2. `f` takes another function `g` as an argument: e.g. `f = fun g => g 0`.
           3. `f` is called by with an argument `g` that does not respect the contract: e.g. `f (fun x => false)`.
     = Either change the contract accordingly, or call `f` with a function that returns a value of the right type.
     = Note: this is an illustrative example. The actual error may involve deeper nested functions calls.
```

Updating the flake lock should do.

**Comment by thufschmitt** [comment]
>  we fixed this bug as of yesterday 

That's what I call reactivity!

Awesome then, I'll just close this
